### ISSUE TITLE
psqldef can't add unique constraint.

### ISSUE THREAD

**Issue by subaru-shoji** [k0kubun/sqldef#173]
## Description
I want to create unique constraint to use upsert statement in postgresql.
However, add constraint statement fails with an error on psqldef.

## --export output
```
-- No table exists --
```

## Input SQL
```sql
create table dummy(
    column_a int not null 
    ,column_b int not null 
    ,column_c int not null
);

alter table dummy add constraint dummy_uniq unique (column_a, column_b);
```
## Current output
```
found syntax error when parsing DDL "alter table dummy add constraint dummy_uniq unique (column_a, column_b)": syntax error at position 33 near 'constraint'
```

## Expected output
```sql
create table dummy(
    column_a int not null 
    ,column_b int not null 
    ,column_c int not null
);

alter table dummy add constraint dummy_uniq unique (column_a, column_b);
```

**Comment by k0kubun** [comment]
It's certainly something psqldef should accept, but I need to change the design a bit to easily deal with variances of some DDLs that have already-supported equivalents. In the meantime, may I ask you to consider using this?

```sql
create table dummy(
    column_a int not null 
    ,column_b int not null 
    ,column_c int not null
);

create unique index dummy_uniq on dummy (column_a, column_b);
```

**Comment by migutw42** [comment]
> It's certainly something psqldef should accept, but I need to change the design a bit to easily deal with variances of some DDLs that have already-supported equivalents. In the meantime, may I ask you to consider using this?
> 
> ```sql
> create table dummy(
>     column_a int not null 
>     ,column_b int not null 
>     ,column_c int not null
> );
> 
> create unique index dummy_uniq on dummy (column_a, column_b);
> ```

Thank you.
I know this sql and I've already tryed to use it to upsert data but upsert doesn't work.

**Comment by k0kubun** [comment]
> I know this sql and I've already tryed to use it to upsert data but upsert doesn't work.

Do you mean upsert works with `alter table dummy add constraint dummy_uniq unique (column_a, column_b);` but doesn't with `create unique index dummy_uniq on dummy (column_a, column_b);`? I was gonna treat those two in the same way, but I guess that's not what you want?

**Comment by k0kubun** [comment]
If that's the case, please let me know the SQL you need to make successful to reproduce the issue. I need a complete reproduction step to think about all feasible approaches.

**Comment by subaru-shoji** [comment]
> If that's the case, please let me know the SQL you need to make successful to reproduce the issue. I need complete reproduction steps, which seem to be not provided for now, to think about all feasible approaches.

I want to run the sql like this.

```sql
INSERT INTO dummy (column_a, column_b, column_c) values (1, 1, 1) ON CONFLICT ON CONSTRAINT dummy_uniq DO UPDATE SET column_a = 2;
```
 this sql requires constraint, not the index.

**Comment by k0kubun** [comment]
That makes sense. Thank you.

It confused me since `pg_indexes` return results as if they are the same thing (`pg_indexes.indexdef` says `CREATE UNIQUE INDEX` too). However, `pg_catalog.pg_constraint` seems to return different things for these, so fixing it seems doable.

I probably can't fix this right away because the situation is a bit complicated around `pg_indexes`, but hopefully I can allocate some time to fix the design around indexes and constraints by the end of this year.

**Comment by k0kubun** [comment]
Fixed the reported test case in v0.11.19.
